schemas/config.json

```diff
@@ -0,0 +1,367 @@
+                    "level": {"$ref": "#/defs/loglevel"},
+                    "handlers": {
+                        "type": "array",
+                        "items": {
+                            "anyOf": [
+                                {"$ref": "#/defs/file_handler"},
+                                {"$ref": "#/defs/filelog_handler"},
+                                {"$ref": "#/defs/graylog_handler"},
+                                {"$ref": "#/defs/stream_handler"},
+                                {"$ref": "#/defs/syslog_handler"}
+                            ]
+                        }
+                    },
+                    "target_systems": {"$ref": "#/defs/system_ref"}
+                },
+                "additionalProperties": false
+            }
+        },
+        "perf_logging": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "level": {"$ref": "#/defs/loglevel"},
+                    "handlers": {
+                        "type": "array",
+                        "items": {
+                            "anyOf": [
+                                {"$ref": "#/defs/file_handler"},
+                                {"$ref": "#/defs/filelog_handler"},
+                                {"$ref": "#/defs/graylog_handler"},
+                                {"$ref": "#/defs/stream_handler"},
+                                {"$ref": "#/defs/syslog_handler"}
+                            ]
+                        }
+                    },
+                    "target_systems": {"$ref": "#/defs/system_ref"}
+                },
+                "additionalProperties": false
+            }
+        },
+        "modes": {
+            "type": "array",
+            "items": {
+            "type": "object",
+                "properties": {
+                    "name": {"type": "string"},
+                    "options": {
+                        "type": "array",
+                        "items": {"type": "string"}
+                    },
+                    "target_systems": {"$ref": "#/defs/system_ref"}
+                },
+                "required": ["name"],
+                "additionalProperties": false
+            }
+        },
+        "general": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "check_search_path": {
+                        "type": "array",
+                        "items": {"type": "string"}
+                    },
+                    "check_search_recursive": {"type": "boolean"},
+                    "target_systems": {"$ref": "#/defs/system_ref"}
+                },
+                "additionalProperties": false
+            }
+        }
+    },
+    "required": ["systems", "environments", "logging", "perf_logging"],
+    "additionalProperties": false,
+    "defaults": {
+        "environments/cc": "cc",
```

Comment:
The defaults are not vanilla Linux. Should we make it 
```
"environments/cc": "cc",
"environments/cxx": "c++",
"environments/ftn": "gfortran",
...
```

Reply:
I agree, but again this is what we have in the documentation. I think this should be a different issue. Can you please open it?

Reply:
@victorusu Can you make a feature request for this?